src/main/java/com/rpg2014/wrappers/SpidermanEC2Wrapper.java

```diff
@@ -233,11 +264,18 @@ private boolean isInstanceStopped(){
         return isDown;
     }
 
-    public Status getStatus(){
+    public Status getInstanceStatus(){
         DescribeInstancesRequest request = DescribeInstancesRequest.builder()
                 .instanceIds(serverDetails.getInstanceId()).build();
-        DescribeInstancesResponse response = ec2Client.describeInstances(request);
-        Status status = Status.of(response.reservations().get(0).instances().get(0).state().code());
-        return status;
+        try {
+            DescribeInstancesResponse response = ec2Client.describeInstances(request);
+            Status status = Status.of(response.reservations().get(0).instances().get(0).state().code());
+            return status;
+        }catch (Ec2Exception e){
+            if(e.getMessage().contains("Invalid id")){
+                return Status.Terminated;
+            }
+        }
+        throw new InternalServerErrorException("Unable to Fetch server status");
```

Comment:
Move this inside the catch so you aren't throwing an error with every getInstanceStatus call

Reply:
We return inside the try block.  This would only get thrown in the case of the error message not containing "invalid id"